trestle/core/commands/author/ssp.py

```diff
@@ -123,46 +126,91 @@ def _init_arguments(self) -> None:
         self.add_argument('-m', '--markdown', help=file_help_str, required=True, type=str)
         output_help_str = 'Name of the output generated json SSP'
         self.add_argument('-o', '--output', help=output_help_str, required=True, type=str)
+        self.add_argument('-r', '--regenerate', action='store_true', help=const.HELP_REGENERATE)
+
+    def _imp_reqs_equivalent(
+        self, imp_reqs: List[ossp.ImplementedRequirement], orig_imp_reqs: List[ossp.ImplementedRequirement]
+    ) -> bool:
+        """
+        Check if imp_reqs are the same except for internal uuids.
+
+        Create copy of each imp_req and set the uuids within it to match the original - then check equality.
+        """
+        if len(as_list(imp_reqs)) == len(as_list(orig_imp_reqs)):
+            for reqs in zip(imp_reqs, orig_imp_reqs):
+                # make a copy of the new imp req
+                tmp_req = copy.deepcopy(reqs[0])
+                if tmp_req.statements is not None and reqs[1].statements is not None:
+                    if len(tmp_req.statements) != len(reqs[1].statements):
+                        return False
+                    tmp_req.uuid = reqs[1].uuid
+                    for stats in zip(tmp_req.statements, reqs[1].statements):
+                        if stats[0].by_components is not None and stats[1].by_components is not None:
+                            if len(stats[0].by_components) != len(stats[1].by_components):
+                                return False
+                            stats[0].uuid = stats[1].uuid
+                            for by_comps in zip(stats[0].by_components, stats[1].by_components):
+                                by_comps[0].uuid = by_comps[1].uuid
+            if tmp_req != reqs[1]:
+                return False
+        return True
 
     def _run(self, args: argparse.Namespace) -> int:
         log.set_log_level_from_args(args)
+        trestle_root = pathlib.Path(args.trestle_root)
 
+        # load the fresh imp_reqs from the markdown
         try:
+            md_path = trestle_root / args.markdown
+        except Exception as e:
```

Comment:
I guess this is because `args.markdown` can be None? As it terminates command execution anyways, wouldn't it be easier to check if the argument is passed and notify the user if it is not?

Reply:
This was a refactoring error.  I have restructured the try block to capture the overall assembly operation.